beater/beater.go

```diff
@@ -130,16 +132,23 @@ type beater struct {
 // Run runs the APM Server, blocking until the beater's Stop method is called,
 // or a fatal error occurs.
 func (bt *beater) Run(b *beat.Beat) error {
-	tracer, tracerServer, err := initTracer(b.Info, bt.config, bt.logger)
+
+	tracerServer, err := newTracerServer(bt.config, b.Instrumentation.Listener())
 	if err != nil {
 		return err
 	}
-	defer tracer.Close()
 
-	runServer := runServer
-	if tracerServer != nil {
-		runServer = runServerWithTracerServer(runServer, tracerServer, tracer)
+	tracer := b.Instrumentation.Tracer()
+	useLegacyTracer := common.MustNewVersion(version.GetDefaultVersion()).LessThan(&common.Version{Major: 8, Minor: 0})
+
+	if !tracer.Active() && useLegacyTracer {
+		tracer, tracerServer, err = initLegacyTracer(b.Info, bt.config, bt.logger)
```

Comment:
Instead of holding onto that old code, can we instead dynamically rename `apm-server.instrumentation` to `instrumentation`, if the latter doesn't exist in the config?

I think we can do this by defining a ConfigOverride in libbeat/cmd/instance.Settings, whose Check method checks for existing "instrumentation", and if not specified sets it to "apm-server.instrumentation"

Reply:
Hmm how would that exactly work?
```
{
			Check: func(cfg *common.Config) bool {
				return cfg.HasField("instrumentation")
			},
			Config: ???
```

`Config` is just a config, not a function, so how do I copy `apm-server.instrumentation`?

Reply:
I meant make `Check` do the modification. It would be a bit of a hack, so it should really be replaced with something more fit for purpose. But in the mean time...

```go
Check: func(cfg *common.Config) bool {
    if !cfg.HasField("instrumentation") {
        // Check if cfg has "apm-server.instrumentation", and rename it to "instrumentation"
    }
    return true
}
```

Reply:
ah, uh. that's quite a hack, yes.. 
ok

Reply:
How about not using `Check` but changing `libbeatConfigOverride` to be a function and taking care of the overwrite there?

Reply:
But how do I pass to a function the original config I need to copy?
Or you mean to change the whole thing in libbeat?

Reply:
I presume @simitt meant changing libbeat. I think that's the right long-term solution. IMO, `ConditionalOverride` should be an implementation of an interface like `ConfigTransformer`:

```go
type ConfigTransformer interface {
    // TransformConfig transforms the provided config, returning a possibly modified config.
    TransformConfig(*common.Config) (*common.Config, error)
}

type ConditionalOverride struct {
    Check OverrideChecker
    Config *common.Config
}

func (o *ConditionalOverride) TransformConfig(in *common.Config) (*common.Config, error) {
    if !o.Check(in) {
        return in, nil
    }
    return common.MergeConfigs(in, o.Config)
}
```

We could then provide an alternative implementation of that interface which does what we need.